src/python/zquantum/core/interfaces/mock_objects.py

```diff
@@ -62,5 +64,40 @@ def get_gradient(self, parameters):
             return self.get_gradients_finite_difference(parameters)
 
 
-def mock_ansatz(parameters):
-    return Circuit(Program(X(0)))
+class MockAnsatz(Ansatz):
+    supported_gradient_methods = ["finite_differences", "parameter_shift_rule"]
+
+    def get_circuit(self):
+        circuit = Circuit()
+        for layer_index in range(self._n_layers):
+            theta = sympy.Symbol("theta" + str(layer_index))
+            for qubit_index in range(self._n_qubits):
+                circuit += Program(RX(theta, qubit_index))
+        return circuit
+
+    def get_gradient_circuits(self):
+        if self.gradient_type == "finite_differences":
+            return [self.get_circuit()]
+        elif self.gradient_type == "parameter_shift_rule":
+            circuit_plus = Circuit()
+            circuit_minus = Circuit()
+            for layer_index in range(self._n_layers):
+                theta = sympy.Symbol("theta" + str(layer_index))
```

Comment:
As above, iteration could be made more readable and consistent.

Reply:
In this case we have nested loop, so I think it's more readable to leave it as it is?

Reply:
@mstechly I added comment that got lost first time I reviewed the code, so probably everything makes more sense now. This iteration will still be nested, but the first loop simplifies to `for theta in self.get_symbols():`.

Reply:
Addressed in [a6862d](https://github.com/zapatacomputing/z-quantum-core/pull/42/commits/a6862dbe5aec14c1bff5d25bb65d7b515067b657).